### PR TITLE
Travis build

### PR DESCRIPTION
**Pull request by aricart**
Added travis build that publishes release artifacts when a tag is created.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  apt:
+    packages:
+      - imagemagick
+      - graphicsmagick
+
+install:
+  - wget https://github.com/gohugoio/hugo/releases/download/v0.31.1/hugo_0.31.1_Linux-64bit.deb
+  - sudo dpkg -i hugo*.deb
+  - npm install
+  - npm install --global gulp-cli
```
Comment (ColinSullivan1) [inline .travis.yml]: Does [Pygments](https://github.com/nats-io/nats-site#install-pygments) need to be installed?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+script:
+  - gulp build
+  - if [[ "$TRAVIS_TAG" != "" ]]; then echo 'Building tag artifact'; fi
```
Comment (ColinSullivan1) [inline .travis.yml]: Can these be in one `if`?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - 3.6.x
+
+addons:
+  apt:
+    packages:
+      - imagemagick
+      - graphicsmagick
+
+install:
+  - wget https://github.com/gohugoio/hugo/releases/download/v0.31.1/hugo_0.31.1_Linux-64bit.deb
```
Comment (ColinSullivan1) [inline .travis.yml]: Is there still an issue with the current release of hugo and nats-site?

File: .travis.yml (hunk 1 above)
Comment (aricart) [reply .travis.yml]: Pygments doesn't even install on travis for some weird error. And current pushes of the site have not used it.

File: .travis.yml (hunk 3 above)
Comment (aricart) [reply .travis.yml]: yes there is.

File: .travis.yml (hunk 2 above)
Comment (aricart) [reply .travis.yml]: They could, but they are not. It is just a debug message, not a complex operation.

File: .travis.yml (hunk 1 above)
Comment (ColinSullivan1) [reply .travis.yml]: OK.

File: .travis.yml (hunk 2 above)
Comment (ColinSullivan1) [reply .travis.yml]: Your call...

**Pull request merged**